src/histolab/slide.py

```diff
@@ -52,13 +52,18 @@ class Slide:
 
     Arguments
     ---------
-    path : str
+    path : Union[str, pathlib.Path]
         Path where the WSI is saved.
-    processed_path : str
+    processed_path : Union[str, pathlib.Path]
         Path where thumbnails and scaled images will be saved to.
     """
 
-    def __init__(self, path: str, processed_path: str) -> None:
+    def __init__(
+        self, path: Union[str, pathlib.Path], processed_path: Union[str, pathlib.Path]
+    ) -> None:
+        if isinstance(path, pathlib.Path):
+            path = str(path)
+
```

Comment:
I think this if is not needed. just cast the path to string in any case and we're done 🤷🏽‍♂️

Reply:
ehm nope
```python
In [1]: str(None)
Out[1]: 'None'
```

Reply:
ok, seems fair so ...why we don't use a ternary operator in the assigment?
```python
self._path = str(path) if isinstance(path, pathlib.Path) else path
```

Reply:
yeah why not